File: BE/src/main/java/com/ssafy/e106/domain/subscription/dto/response/SubscriptionCreateResponse.java
package com.ssafy.e106.domain.subscription.dto.response;

import java.time.LocalDate;
import java.time.LocalDateTime;

import com.ssafy.e106.domain.subscription.entity.Subscription;

public record SubscriptionCreateResponse(
    Long subscriptionId,
    Long userId,
    String subscriptionType,
    Long serviceId,
    Long servicePlanId,
    String bundleCode,
    Integer monthlyPrice,
    Integer originalPrice,
    LocalDate nextBillingDate,
    LocalDateTime createdAt) {

  public static SubscriptionCreateResponse of(Subscription subscription) {
    return new SubscriptionCreateResponse(
        subscription.getSubscriptionId(),
        subscription.getUserId(),
        subscription.getSubscriptionType().name(),
        subscription.getService() == null ? null : subscription.getService().getServiceId(),
        subscription.getServicePlan() == null ? null : subscription.getServicePlan().getServicePlanId(),
        subscription.getBundle() == null ? null : subscription.getBundle().getCode(),
        subscription.getBundle() == null
            ? subscription.getServicePlan().getMonthlyPrice()
            : subscription.getBundle().getMonthlyPrice(),
        subscription.getBundle() == null ? null : subscription.getBundle().getOriginalPrice(),
        subscription.getNextBillingDate(),
        subscription.getCreatedAt());
  }
}

File: FE/app/src/main/java/com/ssafy/e106/data/dto/subscription/SubscriptionCreateResponse.kt
package com.ssafy.e106.data.dto.subscription

import kotlinx.serialization.Serializable

@Serializable
data class SubscriptionCreateResponse(
    val subscriptionId: Long,
    val userId: Long,
    val subscriptionType: String = "SINGLE",
    val serviceId: Long? = null,
    val servicePlanId: Long? = null,
    val bundleCode: String? = null,
    val monthlyPrice: Int,
    val nextBillingDate: String? = null,
    val createdAt: String? = null,
)
